README.md

```diff
@@ -1,14 +1,84 @@
 https://docs.google.com/presentation/d/11a0jibNz0fyUnsWt9fa2MmghHANdHAAABa0TV7EieHs/edit?usp=sharing).
 
+For more information, please check out the [document](https://github.com/ethresearch/sharding-p2p-poc/docs/tree/master/README.md).
+
+## Install
+
+### Prerequisites
+- Make sure you have installed `go` already and set the `$GOPATH` properly
+- `gx` and `gx-go`
+- If you modify `*.proto` files, you will also need `protoc` to compile them to `*.pb.go`
+    - e.g., [Install protoc on mac](https://medium.com/@erika_dike/installing-the-protobuf-compiler-on-a-mac-a0d397af46b8)
+
+### Build
+```bash
+$ go get github.com/ethresearch/sharding-p2p-poc
+$ cd $GOPATH/src/github.com/ethresearch/sharding-p2p-poc
+$ go get ./...  # This will recursively download all of the dependencies
+$ make deps
+$ go build
+```
+
+### Testing
+```bash
+$ go test -v
+```
+
+### Sidenote: version control of packages
+We use [gx](https://github.com/whyrusleeping/gx) as the package management tools for libp2p packages. Running the command `make gx-rw` will rewrite the import statements `github.com/aaa/bbb/ccc` of those packages to `gx/ipfs/xxx/ddd/ccc` in our repository, which makes us use of the specific version of the packages stored on ipfs.
+Command `make gx-uw` will undo the effect of `make gx-rw`.
+
+**NOTE**: (This is a contingency plan) If the packages are `gx-rw`'ed, please make sure it's followed by one more command `make partial-gx-uw` to unwrite certain packages. We apologize for the inconvenience.
 
-# Getting started: Docker dev environment playground
+## Usage
+```bash
+$ ./sharding-p2p-poc --help
+Usage of ./sharding-p2p-poc:
+  -bootnodes string
+    	multiaddresses of the bootnodes
+  -bootstrap
+    	whether to do bootstrapping or not
+  -client
+    	is RPC client or server
+  -port int
+    	port listened by the node for incoming connections (default 10000)
+  -rpcport int
+    	rpc port listened by the rpc server (default 13000)
+  -seed int
+    	set random seed for id generation
+```
+**Note**: `-bootstrap` controls whether to spin up a bootstrap routine, which periodically queries its peers for new peers. There will be no effect if you feed `-bootnodes` without specifying the flag `-bootstrap`.
+### Example
+
+#### Spin up a node with a running RPC server
+
+```bash
+$ ./sharding-p2p-poc -seed=1 -port=10001 -rpcport=13001 -bootstrap -bootnodes=/ip4/127.0.0.1/tcp/5566/ipfs/QmS5QmciTXXnCUCyxud5eWFenUMAmvAWSDa1c7dvdXRMZ7,/ip4/127.0.0.1/tcp/10001/ipfs/QmexAnfpHrhMmAC5UNQVS8iBuUUgDrMbMY17Cck2gKrqeX
+```
+This command spins up a node with seed `0`, listening to new connections at port `10001`, listening to RPC requests at port `13001`, turning on bootstrapping mode with the flag`-bootstrap`, with the bootstrapping nodes `/ip4/127.0.0.1/tcp/5566/ipfs/QmS5QmciTXXnCUCyxud5eWFenUMAmvAWSDa1c7dvdXRMZ7` and `/ip4/127.0.0.1/tcp/10001/ipfs/QmexAnfpHrhMmAC5UNQVS8iBuUUgDrMbMY17Cck2gKrqeX`.
```

Comment:
What do you think about adding a simple explanation on the bootstrapping mode? What's the difference between the bootstrapping mode and specifying bootstrapping nodes?

Reply:
I had some explanation in line 55 for this. Do you think that's enough or I should elaborate more?

Reply:
Sorry I missed that. LGTM.